### PR TITLE
Bugfix/restore standalone balances

### PR DESCRIPTION
**Pull request by piyalbasu**

### CODE REVIEW

File: @shared/api/internal.ts [hunk 1]
```diff
@@ -467,6 +481,156 @@ export const getAccountIndexerBalances = async (
...
+
+  return {
+    balance,
+    decimals,
+    name,
+    symbol,
+  };
+};
+
+export const getAccountBalancesStandalone = async ({
```
Comment (piyalbasu) [inline @shared/api/internal.ts:517]: creating a standalone fetch for balances that combines our old logic of fetching classic assets from a Horizon endpoint and token balances from a Soroban RPC endpoint

File: @shared/api/internal.ts [hunk 2]
```diff
@@ -467,6 +481,156 @@ export const getAccountIndexerBalances = async (
...
+            decimals: await sorobanClientTxBuilder(),
+            symbol: await sorobanClientTxBuilder(),
+          },
+          params,
+        );
+        /* eslint-enable no-await-in-loop */
+
+        const total = new BigNumber(balance);
+
+        tokenBalances[`${symbol}:${tokenId}`] = {
```
Comment (piyalbasu) [inline @shared/api/internal.ts:612]: here I'm conforming the token balance to the shape the indexer provides so we don't run into type issues elsewhere

File: @shared/api/types.ts [hunk 3]
```diff
@@ -177,6 +177,7 @@ export interface SorobanBalance {
   name: string;
   symbol: string;
   decimals: number;
+  token?: { code: string; issuer: { key: string } };
```
Comment (piyalbasu) [inline @shared/api/types.ts:180]: this is added specifically to conform to the indexer's balance response

File: @shared/helpers/soroban/server.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import {
```
Comment (piyalbasu) [inline @shared/helpers/soroban/server.ts:1]: restoring this file

File: @shared/helpers/soroban/token.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
+import {
```
Comment (piyalbasu) [inline @shared/helpers/soroban/token.ts:1]: same. just restoring this file that existed before

File: extension/src/popup/SorobanContext.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+export const SorobanProvider = ({
+  children,
+  pubKey,
+}: {
+  children: React.ReactNode;
+  pubKey: string;
+}) => {
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const { sorobanRpcUrl = "" } = networkDetails;
```
Comment (piyalbasu) [inline extension/src/popup/SorobanContext.tsx:29]: I simplified the logic here as before we were dealing with edge cases where network's didn't have an RPC. I think we can assume everyone will be including an RPC url going forward, but I'll test the fail case of this

File: extension/src/popup/components/manageAssets/AddToken/index.tsx [hunk 7]
```diff
@@ -129,6 +135,26 @@ export const AddToken = () => {
             domain: record.domain,
           })),
         );
+      } else if (isCustomNetwork(networkDetails)) {
+        const name = await getName(
```
Comment (piyalbasu) [inline extension/src/popup/components/manageAssets/AddToken/index.tsx:139]: hit the RPC to find some details about the token ID a user is trying to add

File: extension/src/popup/ducks/transactionSubmission.ts [hunk 8]
```diff
@@ -277,20 +279,36 @@ export const getAccountBalances = createAsyncThunk<
...
-    return thunkApi.rejectWithValue({ errorMessage: e });
-  }
-});
+>(
+  "getAccountBalances",
+  async ({ publicKey, networkDetails, sorobanClient }, thunkApi) => {
+    try {
+      let balances;
+
+      if (isCustomNetwork(networkDetails)) {
```
Comment (piyalbasu) [inline extension/src/popup/ducks/transactionSubmission.ts:291]: splitting the logic between an indexer fetch and a local fetch

**Review by aristidesstaffieri — APPROVED** [review]
